### DISCUSSION TITLE
[Feature Request]Making toolbars more compatible with basic design principles

### DISCUSSION THREAD

**Discussion by Reqrefusion** [FreeCAD/FreeCAD#13775] [Discussions that will turn in to issues but need refining]
The current design is just good enough. To develop this, design principles must be applied by going to the basics. After the internal tabbar, these should slowly become visible.

These principles, created to make the design easier to understand, are must-haves in any design. The application of these principles is non-negotiable. However, how they will be implemented is debatable. I will talk about one that comes to my mind right now.

**Proportion AKA Domination**
The 80-20 rule is a well-known and widely used rule. According to this rule, 80% of the work is done with 20% of the tools. This situation should be reflected in the design. More used tools should be more prominent. Less used items should be hidden to increase visibility. This does not make it difficult to use less frequently used ones, on the contrary, it makes them more understandable since they take up more space when the hiding status is turned off. Let's go through this example that I quickly created to explain it a little better.
![menü](https://github.com/FreeCAD/FreeCAD/assets/39885728/fa8955e2-055a-412e-ab7b-7e8bb50520a7)
In this example, made to show possible placements of icons, you probably see an icon layout that does most of the work and takes up more than 4 times the space of other icons. This is reserved for the tool that is likely to be the most used and/or the tool that describes the relevant group. There may be another area like this. 

On the right, there are places reserved for the tools that are not used as much as the first tool but are within 20%. I have foreseen 3 columns, this may decrease or increase depending on the user or the characteristics of the group.

And finally you see the triangle that stores less used tools. In this now standard design, you can find hidden tools by pressing the triangle. I cannot say that I used negative space, another design principle, very well here.

Additionally, user awareness can be increased by writing the group name below.

In this way, the relevant tools can be used more easily by the user. All tools of the relevant workbench are placed in two lines, ensuring integrity between all workbenches. Especially people who are trying to learn for the first time can be easily guided thanks to this layout. Of course, this picture was an example to explain this principle. Someone who wants to apply this principle can apply it in a different way.

I apologize in advance if this situation has been expressed differently before. Unfortunately, I couldn't find it when I did a keyword search. Yes, this is a controversial issue that should be discussed in the forum, but it should definitely be discussed here as well. I won't move on to other principles so that it won't be too long and I won't be able to fully explain what's on my mind.

**Comment by maxwxyz** [comment]
I'll convert this issue to a discussion.

**Reply by Reqrefusion** [reply]
@maxwxyz thanks for converting. I can't get used to this discussion feature.

**Comment by pierreporte** [comment]
This is basically ribbon UI. While I am a proponent of it for FreeCAD, it would be a very big change that will be met with backlash from a part of the community. If done right, it can mean dislocating workbenches into more than one tab, which itself may need to separate the program into two or more applications (Like Microsoft Office) or workspaces (like Fusion 360). In any case, it’s not happening soon.

**Reply by Reqrefusion** [reply]
This is basically not Ribbon UI. In fact, what is basically known as Ribbon UI is added with Tabbar #12270 . Mentioned here are the basic design principles. These are the first things covered in any design-related course, which you can find a lot of on the internet. Ever Ribbon UI follows these rules because they are universally accepted. I don't want the workbenches to be separated or anything. What I request here is that the icons be placed according to basic design principles. I don't set big goals, things here can be achieved very easily. I just want the extrusion process to be represented with a larger icon in the part design. What you say is debatable, but what I said is a de facto fact that the basic design principles that must be followed in the icon arrangement in workbenches. What can be discussed is not its existence but how it should be. The current situation must definitely change in the future. For this reason, a study must be done soon before 1.0 is released. Please let's not contaminate the topic with big goals.

Thank you very much for joining the discussion.
